R/plot_taxonomy_annotate.R

```diff
@@ -0,0 +1,178 @@
+#' Agglomerate counts of same lineage to specified level of taxonomy.
+#'
+#' @description
+#' Inspired by phyloseq::tax_glom(), this method summarizes k-mer counts from genomes that have the same taxonomy at a user-specified taxonomy rank.
+#' Agglomeration occurs within each sample, meaning the number of k-mers is only summed within each query_name.
+#' This function returns a data frame with the columns `lineage`, `query_name`, and `n_unique_kmers`.
+#'
+#' @param taxonomy_annotate_df Data frame containing outputs from sourmash taxonomy annotate. Can contain results from one or many runs of sourmash taxonomy annotate. Agglomeration occurs within each query.
+#' @param tax_glom_level Character. NULL by default, meaning no agglomeration is done. Valid options are "domain", "phylum", "class", "order", "family", "genus", and "species". When a valid option is supplied, k-mer counts are agglomerated to that level
+#'
+#' @return A data frame.
+#' @export
+#'
+#' @importFrom rlang .data
+#'
+#' @examples
+#' \dontrun{
+#' tax_glom_taxonomy_annotate()
+#' }
+tax_glom_taxonomy_annotate <- function(taxonomy_annotate_df, tax_glom_level = NULL){
```

Comment:
This might just be a personal thing, but the term agglomeration and referring to the function `tax_glom_taxonomy_annotate` seems a little confusing and maybe doesn't clearly convey what this function is doing. I think in R the similar but more known action is aggregating and people might be more familiar with this? Up to you.

Reply:
ah this is interesting, I didn't realize that was already in phyloseq. Maybe somewhere in documentation or just commented out that it's following that convention.